packages/core/src/sync/sync-api.ts

```diff
@@ -0,0 +1,52 @@
+import type { CID } from 'multiformats/cid'
+import { StreamID } from '@ceramicnetwork/streamid'
+
+export type SyncConfig = {
+  /**
+   * Database connection string.
+   */
+  db: string
+}
+
+export interface ISyncApi {
+  startModelSync(startBlock: number, endBlock: number, models: string | string[]): Promise<void>
+  pauseModelSync(models: string | string[]): Promise<void>
+}
+
+interface IpfsService {
+  retrieveFromIPFS(cid: CID | string, path?: string): Promise<any>
+  storeCommit(data: any, streamId?: StreamID): Promise<CID>
+  storeRecord(record: Record<string, unknown>): Promise<CID>
+}
+
+interface Protocol {
```

Comment:
Protocol is a pretty generic name.  Consider something more descriptive or at least a comment explaining what this represents